straxen/plugins/event_processing.py

```diff
@@ -728,31 +730,18 @@ def compute(self, events):
             # use drift time computed using the main S1.
             el_string = peak_type + "s2_interaction_" if peak_type == "alt_" else peak_type
             elife_correction = np.exp(events[f'{el_string}drift_time'] / self.elife)
-
-            cs2_top = cs2_top_wo_elifecorr * elife_correction
-            result[f"{peak_type}cs2_bottom"] = cs2_bottom_wo_elifecorr * elife_correction
-            result[f"{peak_type}cs2"] = cs2_top + result[f"{peak_type}cs2_bottom"]
-
+            cs2_top_wo_timecorr = cs2_top_wo_elifecorr * elife_correction
+            cs2_bottom_wo_timecorr = cs2_bottom_wo_elifecorr * elife_correction
+            result[f"{peak_type}cs2_wo_timecorr"] = cs2_top_wo_timecorr + cs2_bottom_wo_timecorr
+
+            # Correct for SEgain and extraction efficiency
+            seg_ee_corr = (self.se_gain / self.avg_se_gain) * self.rel_extraction_eff
+            result[f"{peak_type}cs2_bottom"] = cs2_bottom_wo_timecorr / seg_ee_corr
+            result[f"{peak_type}cs2"] = cs2_top_wo_timecorr / seg_ee_corr + result[f"{peak_type}cs2_bottom"]
```

Comment:
```suggestion
            result[f"{peak_type}cs2"] = result[f"{peak_type}cs2_wo_timecorr"] / seg_ee_corr 
```
Is a less complicated in my opinion. Follows the same structure as the line above.